app/questionnaire/rules.py

```diff
@@ -30,20 +30,25 @@ def evaluate_goto(goto_rule, metadata, answer_store, group_instance):
     :return:
     """
     if 'when' in goto_rule.keys():
-        when = goto_rule['when']
+        all_conditions_met = True
 
-        if 'id' in when:
-            answer_index = when['id']
-            filtered = answer_store.filter(answer_id=answer_index, group_instance=group_instance)
-            if len(filtered) == 1:
-                return evaluate_rule(when, filtered[0]['value'])
+        for when_rule in goto_rule['when']:
+            condition_met = False
 
-        elif 'meta' in when:
-            key = when['meta']
-            value = get_metadata_value(metadata, key)
-            return evaluate_rule(when, value)
+            if 'id' in when_rule:
+                answer_index = when_rule['id']
+                filtered = answer_store.filter(answer_id=answer_index, group_instance=group_instance)
+                if len(filtered) == 1:
+                    condition_met = evaluate_rule(when_rule, filtered[0]['value'])
 
-        return False
+            elif 'meta' in when_rule:
+                key = when_rule['meta']
+                value = get_metadata_value(metadata, key)
+                condition_met = evaluate_rule(when_rule, value)
+
+            all_conditions_met &= condition_met
```

Comment:
Since the `when` statements are AND conditions, there's no need to do this, you can just return False (or break the loop) as soon as any condition doesn't match i.e. there's no need to go through every `when`.